### DISCUSSION TITLE
Unable to update and commit go.sum files due to private repo dependency

### DISCUSSION THREAD

**Discussion by christing19** [renovatebot/renovate#24605] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.42.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hello, I would like to seek some help from the community with respect to Renovate configurations for a Go repository.

In the current `go.mod` file, we have a dependency that comes from a private repo in a nested subgroup within Gitlab but currently Gitlab’s subgroup feature breaks the functionality of being able to run `go get <private repo>`. [This is an issue that GitLab is aware of](https://gitlab.com/gitlab-org/gitlab/-/issues/36354). 

As such, in our CI pipelines, we have a script to first `git clone` this private repo, and then a `replace` command in the `go.mod` redirects the dependency path to a local path. Because of this private repo dependency and its corresponding `replace` statement, Renovate is showing an "update artifact problem" and as such is unable to update and commit the `go.sum` file to the update MRs, which causes the lint/unit test jobs to fail in our CI pipelines.

Error:
```
Command failed: go get -d -t <private repo>
go: <private repo path> (replaced by ./<local path>): reading <private repo path>/go.mod: open /go.mod: no such file or directory
```

The desired end result is for the `go get` command to skip over the replaced repo, update artifacts + `go.sum` file and commit the `go.sum` along with the `go.mod` changes in the Renovate MR.

### What you tried so far.

Some of the changes we have attempted to resolve this issue include:
- Added `gomodTidyE` to postUpdateOptions 
- Added `gomodMassage` to postUpdateOptions
- Added <private repo> dependency to `ignoreDeps` in renovate.json
- Tried `git clone <private repo>` prior to running `renovate` job in our forked version of the Renovate Runner
- Updated `goGetDirs` in renovate.json to point to all other dependencies (skipping over the <private repo>)

but none of these changes seemed to fix the problem.

If anyone has any advice or insight into how to fix this, please let me know. It would be greatly appreciated! Happy to provide further details as required.

**Comment by candiii-lin** [comment]
Any suggestions? 🙏

**Comment by viceice** [comment]
read the docs?

https://docs.renovatebot.com/getting-started/private-packages/#example-for-gomod
